src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java

```diff
@@ -129,14 +129,27 @@ protected Object readResolve() {
             c.cloud = this;
             // Apply a label that associates an instance configuration with
             // this cloud provider
-            c.appendLabel(CLOUD_ID_LABEL_KEY, String.valueOf(this.name.hashCode()));
+            c.appendLabel(CLOUD_ID_LABEL_KEY, getInstanceUniqueId());
 
             // Apply a label that identifies the name of this instance configuration
             c.appendLabel(CONFIG_LABEL_KEY, c.namePrefix);
         }
         return this;
     }
 
+    public String getInstanceUniqueId() {
```

Comment:
Although use of Javadoc has been inconsistent previously in this project, we really ought to strive to maintain consistent use of it going forward.